libc/benchmarks/gpu/LibcGpuBenchmark.cpp

```diff
@@ -0,0 +1,92 @@
+#include "LibcGpuBenchmark.h"
+#include "src/__support/CPP/algorithm.h"
+#include "src/__support/FPUtil/sqrt.h"
+#include "src/time/gpu/time_utils.h"
+
+namespace LIBC_NAMESPACE {
+namespace benchmarks {
+
+FixedVector<Benchmark *, 64> benchmarks_to_run;
+
+void Benchmark::add_benchmark(Benchmark *benchmark) {
+  benchmarks_to_run.push_back(benchmark);
+}
+
+int Benchmark::run_benchmarks() {
+  for (auto it = benchmarks_to_run.rbegin(), e = benchmarks_to_run.rend();
+       it != e; ++it)
+    (*it)->run();
+  return 0;
```

Comment:
```suggestion
  for (Benchmark *benchmark : benchmarks_to_run)
    benchmark->run();
  return 0;
```
This should have iterators, I guess this being done in reverse order is important? I figured it would just be executing them as they were added. Also, what's the return value for?

Reply:
for some reason `fixedvector.h` only seems to export `rbegin()` and `rend()` and I only see a class for `reverse_iterator` in `__support/CPP/iterator.h` - is there something else I'm missing?

Reply:
So it does, want to make a patch to add the appropriate iterators?

Reply:
I'll give it a shot!